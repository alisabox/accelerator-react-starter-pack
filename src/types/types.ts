import { Action } from 'redux';
import { ThunkAction}  from 'redux-thunk';
import { AxiosInstance } from 'axios';
import { RootState } from '../store/reducers/root-reducer';
import { SeachOptions } from '../const/const';

export type GuitarType = {
  id: number;
  name: string;
  vendorCode: string;
  type: string;
  description: string;
  previewImg: string;
  stringCount: number;
  rating: number;
  price: number;
}

export type CommentType = {
  id: string;
  userName: string;
  advantages: string;
  disadvantages: string;
  comment: string;
  rating: number;
  createAt: Date;
  guitarId: number;
}

export type FilterSettingsType = {
  minPrice: number;
  maxPrice: number;
}

export type GuitarAndCommentsType = GuitarType & {
  comments: CommentType[],
}

export type guitarsStateType = {
  guitarsAndComments: GuitarAndCommentsType[],
  guitarsPerPage: GuitarAndCommentsType[],
  searchResult: GuitarType[],
}

export type filterStateType = {
  filterURLOptions: SeachOptionsType,
  minPrice?: number,
  maxPrice?: number,
}

export type SeachOperatorsBiasedType = {
  [key: string]: string;
}

export type SortOperatorsType = {
  [key: string]: string;
}


export type FilterParamsType = {
  filters: string,
  sort: string,
  order: string,
  pageSearch: string,
  page: string,
}

export type SeachOptionsType = {
  [SeachOptions.PriceMin]: string,
  [SeachOptions.PriceMax]: string,
  [SeachOptions.Acoustic]: boolean,
  [SeachOptions.Electric]: boolean,
  [SeachOptions.Ukulele]: boolean,
  [SeachOptions.FourStrings]: boolean,
  [SeachOptions.SixStrings]: boolean,
  [SeachOptions.SevenStrings]: boolean,
  [SeachOptions.TwelveStrings]: boolean,
}

export type State = RootState;
export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, State, AxiosInstance, Action>;
export type ReselectType<S> = (state: State) => S;
