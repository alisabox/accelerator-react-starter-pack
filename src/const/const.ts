import { SeachOperatorsBiasedType, SortOperatorsType } from '../types/types';

export const APIRoute = {
  Guitars: '/guitars',
  Comments: '/comments',
  Coupons: '/coupons',
  Orders: '/orders',
} as const;

export const AppRoute = {
  Root: '/',
} as const;

export enum ActionType {
  GetGuitarsAndComments = 'MAIN/GET_GUITARS_AND_COMMENTS',
  GetGuitarsPerPage = 'MAIN/GET_GUITARS_PER_PAGE',
  GetSearchResult = 'MAIN/GET_SEARCH_RESULT',
  ClearSearchResult = 'MAIN/CLEAR_SEARCH_RESULT',
  GetFilterResult = 'MAIN/GET_FILTER_RESULT',
  GetFilterURLOptions = 'MAIN/GET_FILTER_URL_OPTIONS',
  GetMinPrice = 'MAIN/GET_MIN_PRICE',
  GetMaxPrice = 'MAIN/GET_MAX_PRICE',
}

export enum CatalogSort {
  Price = 'price',
  Rating = 'rating',
  None = 'none',
}

export enum CatalogSortOrder {
  Down = 'desc',
  Up = 'asc',
  None = 'none',
}

export const CatalogSortOperators: SortOperatorsType = {
  [CatalogSort.Price]: '_sort=price',
  [CatalogSort.Rating]: '_sort=rating',
  [CatalogSortOrder.Down]: '_order=desc',
  [CatalogSortOrder.Up]: '_order=asc',
} as const;

export const SeachOperators = {
  MinPrice: 'price_gte=',
  MaxPrice: 'price_lte=',
  TypeAcoustic: 'type=acoustic',
  TypeElectric: 'type=electric',
  TypeUkulele: 'type=ukulele',
  FourStrings: 'stringCount=4',
  SixStrings: 'stringCount=6',
  SevenStrings: 'stringCount=7',
  TwelveStrings: 'stringCount=12',
} as const;

export enum SeachOptions {
  PriceMin = 'priceMin',
  PriceMax = 'priceMax',
  Acoustic = 'acoustic',
  Electric = 'electric',
  Ukulele = 'ukulele',
  FourStrings = '4-strings',
  SixStrings = '6-strings',
  SevenStrings = '7-strings',
  TwelveStrings = '12-strings',
}

export const SeachOperatorsBiased: SeachOperatorsBiasedType = {
  [SeachOptions.PriceMin]: SeachOperators.MinPrice,
  [SeachOptions.PriceMax]: SeachOperators.MaxPrice,
  [SeachOptions.Acoustic]: SeachOperators.TypeAcoustic,
  [SeachOptions.Electric]: SeachOperators.TypeElectric,
  [SeachOptions.Ukulele]: SeachOperators.TypeUkulele,
  [SeachOptions.FourStrings]: SeachOperators.FourStrings,
  [SeachOptions.SixStrings]: SeachOperators.SixStrings,
  [SeachOptions.SevenStrings]: SeachOperators.SevenStrings,
  [SeachOptions.TwelveStrings]: SeachOperators.TwelveStrings,
} as const;

export const NUMBER_OF_CARDS = 9;

export const ERROR_MESSAGE = 'Не удалось загрузить данные каталога. Попробуйте еще раз позже';

export enum EmptyCatalogMessage {
  Loading = 'Загружаем...',
  ConnectionError = 'Не удалось загрузить каталог. Проверьте подключение к интернету.',
}
