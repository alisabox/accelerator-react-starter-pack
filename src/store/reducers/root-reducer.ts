import { combineReducers } from 'redux';
import { filterReducer } from './filter/filter-reducer';
import { guitarsReducer } from './guitars/guitars-reducer';

export enum NameSpace {
  Guitars = 'Guitars',
  Filter = 'Filter',
}

export const rootReducer = combineReducers({
  [NameSpace.Guitars]: guitarsReducer,
  [NameSpace.Filter]: filterReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
