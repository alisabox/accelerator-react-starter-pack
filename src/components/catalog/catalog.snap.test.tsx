import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { createMemoryHistory } from 'history';
import { createAPI } from '../../services/api';
import thunk, { ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { State } from '../../types/types';
import Catalog from './catalog';
import { guitars } from '../../mocks/mocks';
import { NameSpace } from '../../store/reducers/root-reducer';
import { SeachOptions } from '../../const/const';

const history = createMemoryHistory();
const mockData = guitars;
const mockDataPerPage = guitars.slice(0, 10);

const api = createAPI();
const middlewares = [thunk.withExtraArgument(api)];
const mockStore = configureMockStore<
    State,
    Action,
    ThunkDispatch<State, typeof api, Action>
  >(middlewares);

const store = mockStore({
  [NameSpace.Guitars]: {
    guitarsAndComments: mockData,
    guitarsPerPage: mockDataPerPage,
  },
  [NameSpace.Filter]: {
    filterURLOptions: {
      [SeachOptions.PriceMin]: '',
      [SeachOptions.PriceMax]: '',
      [SeachOptions.Acoustic]: false,
      [SeachOptions.Electric]: false,
      [SeachOptions.Ukulele]: false,
      [SeachOptions.FourStrings]: false,
      [SeachOptions.SixStrings]: false,
      [SeachOptions.SevenStrings]: false,
      [SeachOptions.TwelveStrings]: false,
    },
  },
});
describe('Component: Catalog', () => {
  test('should render correctly vs. snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <Catalog />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
