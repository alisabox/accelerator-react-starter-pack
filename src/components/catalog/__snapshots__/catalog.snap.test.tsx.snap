// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Catalog should render correctly vs. snapshot 1`] = `
<div>
  <main
    class="page-content"
  >
    <div
      class="container"
    >
      <h1
        class="page-content__title title title--bigger"
      >
        Каталог гитар
      </h1>
      <ul
        class="breadcrumbs page-content__breadcrumbs"
      >
        <li
          class="breadcrumbs__item"
        >
          <a
            class="link"
            href="./main.html"
          >
            Главная
          </a>
        </li>
        <li
          class="breadcrumbs__item"
        >
          <a
            class="link"
            href="/#"
          >
            Каталог
          </a>
        </li>
      </ul>
      <div
        class="catalog"
      >
        <form
          class="catalog-filter"
        >
          <h2
            class="title title--bigger catalog-filter__title"
          >
            Фильтр
          </h2>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Цена, ₽
            </legend>
            <div
              class="catalog-filter__price-range"
            >
              <div
                class="form-input"
              >
                <label
                  class="visually-hidden"
                >
                  Минимальная цена
                </label>
                <input
                  id="priceMin"
                  name="priceMin"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="form-input"
              >
                <label
                  class="visually-hidden"
                >
                  Максимальная цена
                </label>
                <input
                  id="priceMax"
                  name="priceMax"
                  type="number"
                  value=""
                />
              </div>
            </div>
          </fieldset>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Тип гитар
            </legend>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="acoustic"
                name="acoustic"
                type="checkbox"
              />
              <label
                for="acoustic"
              >
                Акустические гитары
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="electric"
                name="electric"
                type="checkbox"
              />
              <label
                for="electric"
              >
                Электрогитары
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="ukulele"
                name="ukulele"
                type="checkbox"
              />
              <label
                for="ukulele"
              >
                Укулеле
              </label>
            </div>
          </fieldset>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Количество струн
            </legend>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="4-strings"
                name="4-strings"
                type="checkbox"
              />
              <label
                for="4-strings"
              >
                4
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="6-strings"
                name="6-strings"
                type="checkbox"
              />
              <label
                for="6-strings"
              >
                6
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="7-strings"
                name="7-strings"
                type="checkbox"
              />
              <label
                for="7-strings"
              >
                7
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                id="12-strings"
                name="12-strings"
                type="checkbox"
              />
              <label
                for="12-strings"
              >
                12
              </label>
            </div>
          </fieldset>
        </form>
        <div
          class="catalog-sort"
        >
          <h2
            class="catalog-sort__title"
          >
            Сортировать:
          </h2>
          <div
            class="catalog-sort__type"
          >
            <button
              aria-label="по цене"
              class="catalog-sort__type-button "
              id="price"
            >
              по цене
            </button>
            <button
              aria-label="по популярности"
              class="catalog-sort__type-button "
              id="rating"
            >
              по популярности
            </button>
          </div>
          <div
            class="catalog-sort__order"
          >
            <button
              aria-label="По возрастанию"
              class="catalog-sort__order-button catalog-sort__order-button--up "
              id="asc"
            />
            <button
              aria-label="По убыванию"
              class="catalog-sort__order-button catalog-sort__order-button--down "
              id="desc"
            />
          </div>
        </div>
        <div
          class="cards catalog__cards"
        >
          <div
            class="product-card"
          >
            <img
              alt="Честер Bass"
              height="190"
              src="img/content/guitar-1.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Честер Bass
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                17500
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="CURT Z300"
              height="190"
              src="img/content/guitar-8.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                CURT Z300
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                29500
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Roman LX"
              height="190"
              src="img/content/guitar-6.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Roman LX
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                6800
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="CURT T300"
              height="190"
              src="img/content/guitar-3.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                CURT T300
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                30000
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Dania Super"
              height="190"
              src="img/content/guitar-4.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Dania Super
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                3500
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Честер WX"
              height="190"
              src="img/content/guitar-2.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Честер WX
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                15300
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Dania VX"
              height="190"
              src="img/content/guitar-5.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Dania VX
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                2200
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Честер Plus"
              height="190"
              src="img/content/guitar-7.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Честер Plus
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                30000
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
          <div
            class="product-card"
          >
            <img
              alt="Виолана 300"
              height="190"
              src="img/content/guitar-1.jpg"
              width="75"
            />
            <div
              class="product-card__info"
            >
              <div
                aria-hidden="true"
                class="rate product-card__rate"
              >
                <span
                  class="visually-hidden"
                >
                  Рейтинг:
                </span>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-full-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  height="11"
                  width="12"
                >
                  <use
                    xlink:href="#icon-star"
                  />
                </svg>
                <span
                  class="rate__count"
                >
                  0
                </span>
                <span
                  class="rate__message"
                />
              </div>
              <p
                class="product-card__title"
              >
                Виолана 300
              </p>
              <p
                class="product-card__price"
              >
                <span
                  class="visually-hidden"
                >
                  Цена:
                </span>
                1700
                 ₽
              </p>
            </div>
            <div
              class="product-card__buttons"
            >
              <a
                class="button button--mini"
                href="/#"
              >
                Подробнее
              </a>
              <a
                class="button button--red button--mini button--add-to-cart"
                href="/#"
              >
                Купить
              </a>
            </div>
          </div>
        </div>
        <div
          class="pagination page-content__pagination"
        >
          <ul
            class="pagination__list"
          >
            <li
              class="pagination__page pagination__page--active"
            >
              <a
                class="link pagination__page-link"
                href="1"
              >
                1
              </a>
            </li>
            <li
              class="pagination__page null"
            >
              <a
                class="link pagination__page-link"
                href="2"
              >
                2
              </a>
            </li>
            <li
              class="pagination__page null"
            >
              <a
                class="link pagination__page-link"
                href="3"
              >
                3
              </a>
            </li>
            <li
              class="pagination__page pagination__page--next"
              id="next"
            >
              <a
                class="link pagination__page-link"
                href="2"
              >
                Далее
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
</div>
`;
